### PR TITLE
op-geth: update to include v1.13.2 and v1.13.3 geth changes

### PR DESCRIPTION
**Pull request by protolambda**
Depends on https://github.com/ethereum-optimism/op-geth/pull/153

The `Address.Hash()`  util was removed, but otherwise no major interface changes.

### CODE REVIEW

**Review by ajsutton — APPROVED** [review]

**Comment by protolambda** [comment]
Still need to update the `go.mod` to point to canonical commit instead of the now merged PR. Also syncing other op-geth fixes

**Review by trianglesphere — APPROVED** [review]

**Comment by ajsutton** [comment]
> Still need to update the go.mod to point to canonical commit instead of the now merged PR. Also syncing other op-geth fixes

I had pointed it to the canonical commit in https://github.com/ethereum-optimism/optimism/pull/7645/commits/22b8ee122a13eb65130d2ec0116131c1b7311ec7 - I just took over your PRs yesterday to keep things moving. :) Should have left a comment here though sorry.

**Comment by ajsutton** [comment]
But yeah we want to pull in the new cherry pick as well, just merged that so will update the go.mod again to point to the new latest optimism.

File: go.mod [hunk 1]
```diff
@@ -208,7 +209,7 @@ require (
 	rsc.io/tmplfunc v0.0.3 // indirect
 )
 
-replace github.com/ethereum/go-ethereum v1.13.1 => github.com/ethereum-optimism/op-geth v1.101301.0-rc.2.0.20231002141926-1e6910b91798
+replace github.com/ethereum/go-ethereum v1.13.3 => github.com/ethereum-optimism/op-geth v1.101301.2-0.20231018201518-63125bd85c80
```
Comment (protolambda) [inline go.mod]: Can confirm that this is the correct commit to pull in

File: go.mod [hunk 2]
```diff
@@ -8,8 +8,8 @@ require (
 	github.com/btcsuite/btcd/chaincfg/chainhash v1.0.1
 	github.com/decred/dcrd/dcrec/secp256k1/v4 v4.2.0
 	github.com/ethereum-optimism/go-ethereum-hdwallet v0.1.3
-	github.com/ethereum-optimism/superchain-registry/superchain v0.0.0-20231001123245-7b48d3818686
-	github.com/ethereum/go-ethereum v1.13.1
+	github.com/ethereum-optimism/superchain-registry/superchain v0.0.0-20231018164214-046f42968aec
```
Comment (protolambda) [inline go.mod]: Also good to see this updated to match op-geth update :+1:

**Comment by protolambda** [comment]
CI is failing due to a repo-wide CI issue with the builder image, this is being fixed. This PR can rerun CI after that + then be merged.

File: go.mod (hunk 2 above)
Comment (ajsutton) [reply go.mod]: Interestingly, that wasn't me. go mod tidy did it automatically which is a bonus.

**Pull request merged**